### DISCUSSION TITLE
DateTime.AddWeeks

### DISCUSSION THREAD

**Discussion by hrrrrustic** [dotnet/runtime#57734] [Ideas]
Is there any reason why there is no `AddWeeks()` method on DateTime? I can't find any issues/discussions about it
I thought it always had been there, but I typed "AddW" today and didn't find it 😄 

I understand that it's simply `AddDays(smth * 7)`, but 
* Same can be done for hours for example, just `AddMinutes(smth * 60)`
* Week is kinda basic block of dates, seems logical to have separate api for it
* Also `AddWeeks` method exists in [Calendar class](https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Private.CoreLib/src/System/Globalization/Calendar.cs#L248) and it seems a little bit inconsistently

**Comment by Clockwork-Muse** [comment]
> I understand that it's simply `AddDays(smth * 7)`

... `Calendar` has one because weeks aren't always 7 days (although since `DateTime(Offset)` is essentially the ISO calendar, the shortcut is probably okay).

**Reply by hrrrrustic** [reply]
Yeah, that's true, but it looks strange that more complicated type has such an easy method, while a simple `DateTime` that is more widely used not

**Reply by Clockwork-Muse** [reply]
>  has such an easy method

This is false, as week length can vary throughout a "year" (or other measure), or whether a date is even considered part of a normal "week".
The somewhat satirical Discordian calendar considers the equivalent to February 29th to be an "outside the week" day (IIRC, it actually splits the week in two) - as such, `AddWeeks` would skip the day, and _starting_ on that date raises an interesting question....
